backend/FriendsList/friendsRouter.js

```diff
@@ -0,0 +1,230 @@
+        const friendArray = [];
+
+        for (const friend of allAcceptedFriends){
+            console.log('Friend:', friend);
+            console.log('friend id:', friend.friend_id)
+            const friendInfo = await User.findOne({ _id: friend.friend_id }, { username: 1, fname: 1, lname: 1, _id: 1 });
+            console.log(friendInfo);
+            if (friendInfo) {
+                friendArray.push(friendInfo);
+            }
+        }
+        console.log(allAcceptedFriends);
+        console.log(friendArray);
+
+        // Should be sending the friendArray - to get their info sent through
+        res.status(200).json({ allAcceptedFriends });
+        
+    } catch (err) {
+        console.log('Error getting all accepted friends of user', err);
+        res.status(500).json({ message: 'internal server error' })
+    }
+})
+
+// Get all pending friends of user
+router.get('/get/pending', async (req, res) => {
+    const userId = req.query.userId;
+
+    try {
+        const allPendingFriends = await Friend.find({ user_id: userId, status: "pending" })
+
+        // To be able to get the friend's name
+        const pendingfriendArray = [];
+
+        for (const friend of allPendingFriends){
+            console.log('Friend:', friend);
+            console.log('friend id:', friend.friend_id)
+            const friendInfo = await User.findOne({ _id: friend.friend_id }, { username: 1, fname: 1, lname: 1, _id: 1 });
+            console.log(friendInfo);
+            if (friendInfo) {
+                pendingfriendArray.push(friendInfo);
+            }
+        }
+        console.log(pendingfriendArray);
+        console.log(allPendingFriends);
+
+        res.status(200).json({ allPendingFriends });
+    } catch (err) {
+        console.log('Error getting all pending friends of user', err);
+        res.status(500).json({ message: 'internal server error' })
+    }
+})
+
+// Get all pending friend requests
+router.get('/get/requests', async (req, res) => {
+    const userId = req.query.userId;
+
+    try {
+        const friendRequests = await Friend.find({ friend_id: userId, status: "pending" })
+
+        // To be able to get the friend's name
+        const friendRequestArray = [];
+
+        for (const friend of friendRequests){
+            console.log('Friend:', friend);
+            console.log('friend id:', friend.friend_id)
+            const friendInfo = await User.findOne({ _id: friend.friend_id }, { username: 1, fname: 1, lname: 1, _id: 1 });
+            console.log(friendInfo);
+            if (friendInfo) {
+                friendRequestArray.push(friendInfo);
+            }
+        }
+        console.log(friendRequestArray);
+        console.log("Pending friend requests:", friendRequests);
+
+        res.status(200).json({ friendRequests });
+    } catch (err) {
+        console.log('Error getting all pending friend requests of user:', err);
+        res.status(500).json({ message: 'internal server error' })
+    }
+})
```

Comment:
Function performs the same functionality as `router.get('/get/pending', async (req, res) => {`